From 5e61782c36021b83abafec9ad5c419dcb92e4195 Mon Sep 17 00:00:00 2001

Require Tcl-8.5- instead of just Tcl-8.5 to allow for Tcl-9.0

--- engine/tcl/engine.c
+++ engine/tcl/engine.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2010-2015,2019,2022-2023 LAAS/CNRS
+ * Copyright (c) 2010-2015,2019,2022-2023, 2025 LAAS/CNRS
  * All rights reserved.
  *
  * Redistribution  and  use  in  source  and binary  forms,  with  or  without
@@ -196,7 +196,7 @@ engine_invoke(const char *tmpl, int argc, const char * const *argv)
 
   /* initialize interpreter */
   if (Tcl_Init(interp) == TCL_ERROR) goto error;
-  if (!Tcl_PkgRequire(interp, "Tcl", "8.5", 0))
+  if (!Tcl_PkgRequire(interp, "Tcl", "8.5-", 0))
     goto error;
 
   /* define helper procs */
diff --git engine/tcl/engine.tcl engine/tcl/engine.tcl
index b7547fa..8de8d3d 100644
--- engine/tcl/engine.tcl
+++ engine/tcl/engine.tcl
@@ -21,7 +21,7 @@
 #
 #                                           Anthony Mallet on Fri Jan  8 2010
 #
-package require Tcl 8.5
+package require Tcl 8.5-
 
 # tcl<8.6 is missing this bit, provide a good-enough replacement as long as
 # only 8.5 is required.
diff --git engine/tcl/language-c.tcl engine/tcl/language-c.tcl
index bda835c..6e9e9cb 100644
--- engine/tcl/language-c.tcl
+++ engine/tcl/language-c.tcl
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2010-2018,2020,2022-2024 LAAS/CNRS
+# Copyright (c) 2010-2018,2020,2022-2025 LAAS/CNRS
 # All rights reserved.
 #
 # Redistribution  and  use  in  source  and binary  forms,  with  or  without
@@ -21,7 +21,7 @@
 #
 #                                           Anthony Mallet on Tue Jan 26 2010
 #
-package require Tcl 8.5
+package require Tcl 8.5-
 
 namespace eval language::c {
 
diff --git engine/tcl/merge.tcl engine/tcl/merge.tcl
index c60b2cd..3f6dcd2 100644
--- engine/tcl/merge.tcl
+++ engine/tcl/merge.tcl
@@ -22,7 +22,7 @@
 #                                           Anthony Mallet on Wed Mar 30 2011
 #
 
-package require Tcl 8.5
+package require Tcl 8.5-
 
 namespace eval merge {
 
